room/views.py

```diff
@@ -7,88 +7,110 @@
 
 from .forms import AddRoom, EmailInvite, AcceptInvitation, EditRoom
 from .models import Room, Invitation
+from users.models import CustomUser
 from spotify.models import Playlist
 from .email_utils import send_invitation, send_invitation_accepted
 
 
 @login_required
 def add_room(request):
-    if request.method == 'POST':
-            form = AddRoom(request.POST)
+    if request.method == "POST":
+        form = AddRoom(request.POST)
 
-            if form.is_valid():
-                title = request.POST.get('title')                
-                room = Room.objects.create(title=title, created_by=request.user)
-                room.members.add(request.user)
-                room.save()                             
-                request.user.active_room_id = room.id
-                # This would probably be a good place to get authorization
-                # from Spotify - code, token, refresh token. Or do it later on?
-                return redirect('grooveboard')
+        if form.is_valid():
+            title = request.POST.get("title")
+            room = Room.objects.create(title=title, created_by=request.user)
+            room.members.add(request.user)
+            room.save()
+            request.user.active_room_id = room.id
+            # This would probably be a good place to get authorization
+            # from Spotify - code, token, refresh token. Or do it later on?
+            return redirect("grooveboard")
```

Comment:
This is unstructured code. It's easily fixed.
```python
@login_required
def add_room(request):
    form = request.method == "POST" and AddRoom(request.POST)
    if form and form.is_valid():
        title = request.POST.get("title")
        room = Room.objects.create(title=title, created_by=request.user)
        room.members.add(request.user)
        room.save()
        request.user.active_room_id = room.id
        # This would probably be a good place to get authorization
        # from Spotify - code, token, refresh token. Or do it later on?
        return redirect("grooveboard")
    else:
        return render(request, "add_room.html", {"form": AddRoom()})
```

Reply:
Agreed. Again this is black working its magic. Here's the original:
```
@login_required
def add_room(request):
    if request.method == 'POST':
            form = AddRoom(request.POST)

            if form.is_valid():
                title = request.POST.get('title')                
                room = Room.objects.create(title=title, created_by=request.user)
                room.members.add(request.user)
                room.save()                             
                request.user.active_room_id = room.id
                # This would probably be a good place to get authorization
                # from Spotify - code, token, refresh token. Or do it later on?
                return redirect('grooveboard')
    else:

        form = AddRoom()

    return render(request, 'add_room.html', {'form': form})
```

Now that I have seen it (black) in action over a number of files and it has, for example, mangled the urls.py file so it is difficult to read and I wrote all of those paths in it! I personally would never use it.  
I'm off to find out if there is a reverse black option : )

Reply:
Agreed.  
You will see my previous code structure in all the view functions that I wrote to deal with form data being posted. I pretty much copied and pasted the function structure from the Django documentation as I was teaching myself about forms. Here's the Django docs example:
```
from django.http import HttpResponseRedirect
from django.shortcuts import render

from .forms import NameForm

def get_name(request):
    # if this is a POST request we need to process the form data
    if request.method == 'POST':
        # create a form instance and populate it with data from the request:
        form = NameForm(request.POST)
        # check whether it's valid:
        if form.is_valid():
            # process the data in form.cleaned_data as required
            # ...
            # redirect to a new URL:
            return HttpResponseRedirect('/thanks/')

    # if a GET (or any other method) we'll create a blank form
    else:
        form = NameForm()
    return render(request, 'name.html', {'form': form})
```

Reply:
To be totally fair, this is because there is a (well-known) problem in writing structured code, owing to the fact that the `if-then-else` is actually a bit inadequate and can lead to code-duplication. And this particular method is a case in point. As an aside, in [Nutmeg](/Spicery/Nutmeg) we plan to add a [thenif](https://github.com/Spicery/Nutmeg/issues/64) keyword to address this.